planner: plan cache support subquery (#13228)
pingcap/docs-cn: sql-prepared-plan-cache.md

- `SELECT`、`UPDATE`、`INSERT`、`DELETE`、`Union`、`Intersect`、`Except` 以外的 SQL 语句；
- 访问分区表、临时表或访问表中包含生成列的查询；

> - 查询中包含非关联子查询或无法解关联的关联子查询；

Reviewer: 需要举一个例子来说明下 关联子查询 和 非关联子查询

Author: ```suggestion
- 查询中包含非关联子查询或无法解关联的关联子查询，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` 和 `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` 分别是无法缓存的非关联子查询和关联子查询
```

> - 查询中包含非关联子查询或无法解关联的关联子查询，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` 和 `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` 分别是无法缓存的非关联子查询和关联子查询

Author: ```suggestion
- 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` 
- 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` 
```

- `SELECT`、`UPDATE`、`INSERT`、`DELETE`、`Union`、`Intersect`、`Except` 以外的 SQL 语句；
- 访问分区表、临时表或访问表中包含生成列的查询；

> - 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` 

Author: ```suggestion
- 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` ；
```

> - 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` 

Author: ```suggestion
- 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` ；
```

> - 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)`；
> - 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)`；

Reviewer 2: ```suggestion
- 查询中包含非关联子查询，例如 `SELECT * FROM t1 WHERE t1.a > (SELECT 1 FROM t2 WHERE t2.b < 1)`；
- 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如 `SELECT * FROM t1 WHERE t1.a > (SELECT a FROM t2 WHERE t1.b > t2.b)`；
```

> - 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)` ；
> - 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)` ；

Author: ```suggestion
- 查询中包含非关联子查询的，例如 `select * from t1 where t1.a > (select 1 from t2 where t2.b < 1)`；
- 执行计划中带有 `PhysicalApply` 算子的关联子查询，例如  `select * from t1 where t1.a > (select a from t2 where t1.b > t2.b)`；
```
